### ISSUE TITLE
When editing an entry, the text presented is an older version than the one in display mode

### ISSUE THREAD

**Issue by rickyweb** [Journaly/journaly#416]
|                    |                      |
| ------------------ | -------------------- |
| repro              | [link](https://todo) |
| `node` version     | `...`                |
| OS and its version | `...`                |

#### Description
I've come across a bug when editing my entry: https://journaly.com/post/900
After getting some comments, I've partly edited it, saved it (let's call it version A), then returned to it, made some more changes and saved it again (version B). Now the text shown is version B. However, if I click edit again, the version in the editor is the version A. I've tried clearing the cache and also logging out and in again, but there's no change.

Later I edited on another device and this time it worked.

Robin wrote to me: "This is actually a bug we've had but it's been very hard to track down and reproduce as it doesn't seem to happen consistently. "

So below are the steps I can recollect, with no guarantee that it's possible to reproduce it

#### Repro Steps
Steps 1-2, 4-9 on desktop computer, step 10 on a mobile device, everything in the latest Firefox
I don't know if that matters...

1. User A creates a new entry and saves it as a draft (version 0.1)
2. User A updates the draft and saves it (version 1)
3. User B starts to comment some parts of the text
4. While User B is still making inline comments, user A edits parts of the text affected by the comments
5. User A saves his text (version 2)
6. User A now sees version 1 displayed with the comment of user B (the comment was not in the outdated section)
7. User A edits the text yet again and sees the correct version 2 in his edit window
8. User A saves the text again (version 3)
9. User A sees version 3 displayed, but when going into edit mode again, he sees version 2.
10. User A later loads the page on his mobile device and everything works correctly (version 3 shown and edited) Not sure whether the change of device or the timespan between steps 9 and 10 are the reason why the issue disappeared.

**Comment by robin-macpherson** [comment]
Thanks so much for this @rickyweb ! We'll make some time to look into this and let you know if we need more information soon 🙏🏼

**Comment by rickyweb** [comment]
@robin-macpherson, I've had the bug again today. It is not related to the comments as I've written yesterday. The issue is more likely to be located in the application cache (not the browser cache), the feature which will allow you to start writing something, and even without saving, going to another page inside journaly and then go back to your post without losing anything. That cache just needs to be deleted whenever one saves the entry.

**Comment by robin-macpherson** [comment]
@rickyweb thanks, yes we have some code that is supposed to clear that whenever a post is saved or published, but it seems to not be working properly. The strange thing is that the behavior does not seem to be happening consistently. Are you familiar with how to clear your local storage? If so, just as a simple test could you try clearing your local storage and letting us know if this changes anything?

Let me know if you need any help doing this. Thanks again for helping us investigate!

**Comment by rickyweb** [comment]
Ok, yes! I can reproduce it now. The issue is that when saving, it deletes the local storage entries for langId and title, but not for body.
When I opened the local storage browser of my Mozilla Firefox 84.0.1, I saw all body entries of my previous posts.
The funny thing is when saving, all three entries disappear from the local storage browser, but when I close and open it again, then the body is back. :(
